lib/Controller/RoomController.php

```diff
@@ -525,14 +525,19 @@ public function removeFromFavorites($token) {
 	public function renameRoom($token, $roomName) {
 		try {
```

Comment:
The same `try { ... }` block is in multiple functions now, can this somehow be refactored in a separate function to avoid code duplication?

Maybe a new decorator similar to `@NoAdminRequired`, e.g. `@TalkModeratorRequired` that performs the code and checks for `hasModeratorPermissions()`.

Reply:
yeah, something for later.
I'm not sure if apps can already register middleware annotations.

Reply:
But we would then also get the room twice etc, which is kind of dislike about the annotations.